File: CS/WindowsApplication1/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraTreeList.Columns;
using DevExpress.XtraTreeList.Nodes.Operations;
using DevExpress.XtraTreeList.Nodes;

namespace WindowsApplication1
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        	private void treeList1_KeyPress(object sender, System.Windows.Forms.KeyPressEventArgs e)
		{
			if(e.KeyChar > 31) { // a letter
				TreeListOperationFindNodeByText op;
				op = new TreeListOperationFindNodeByText(treeList1.FocusedColumn, e.KeyChar.ToString());
				treeList1.NodesIterator.DoOperation(op);
				if(op.Node != null)
					treeList1.FocusedNode = op.Node;
			}
		}
	}

	internal class TreeListOperationFindNodeByText : TreeListOperation 
	{
		private TreeListNode foundNode;
		private TreeListColumn column;
		private string substr;
		public TreeListOperationFindNodeByText(TreeListColumn column, string substr) 
		{ 
			this.foundNode = null;	
			this.column = column;
			this.substr = substr;
		}
		public override void Execute(TreeListNode node)
		{
			string s = node[column].ToString();
			if(s.StartsWith(substr))
				this.foundNode = node;
		}
		public override bool NeedsVisitChildren(TreeListNode node) { return foundNode == null; }
		public TreeListNode Node { get { return foundNode; } }
	}

}

File: VB/WindowsApplication1/Form1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms
Imports DevExpress.XtraTreeList.Columns
Imports DevExpress.XtraTreeList.Nodes.Operations
Imports DevExpress.XtraTreeList.Nodes

Namespace WindowsApplication1
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

			Private Sub treeList1_KeyPress(ByVal sender As Object, ByVal e As System.Windows.Forms.KeyPressEventArgs) Handles treeList1.KeyPress
            If Asc(e.KeyChar) > 31 Then ' a letter
                Dim op As TreeListOperationFindNodeByText
                op = New TreeListOperationFindNodeByText(treeList1.FocusedColumn, e.KeyChar.ToString())
                treeList1.NodesIterator.DoOperation(op)
                If op.Node IsNot Nothing Then
                    treeList1.FocusedNode = op.Node
                End If
            End If
			End Sub
	End Class

	Friend Class TreeListOperationFindNodeByText
		Inherits TreeListOperation
		Private foundNode As TreeListNode
		Private column As TreeListColumn
		Private substr As String
		Public Sub New(ByVal column As TreeListColumn, ByVal substr As String)
			Me.foundNode = Nothing
			Me.column = column
			Me.substr = substr
		End Sub
		Public Overrides Sub Execute(ByVal node As TreeListNode)
			Dim s As String = node(column).ToString()
			If s.StartsWith(substr) Then
				Me.foundNode = node
			End If
		End Sub
		Public Overrides Function NeedsVisitChildren(ByVal node As TreeListNode) As Boolean
			Return foundNode Is Nothing
		End Function
		Public ReadOnly Property Node() As TreeListNode
			Get
				Return foundNode
			End Get
		End Property
	End Class

End Namespace